### PR TITLE
Precommit updates

### PR DESCRIPTION
**Pull request by adamgayoso**
Use scverse template pre commit. We can now likely remove the static code checks from the regular test workflow.

### CODE REVIEW

**Comment by adamgayoso** [comment]
@martinkim0 can you help resolve the pre-commit issues in this PR? Feel free to add commits directly to this PR

File: .flake8 [hunk 1]
```diff
@@ -1,5 +1,66 @@
...
-
+ignore =
+    # Unnecessary dict call - rewrite as a literal.
+    C408
+    # Missing docstring in public class
+    D101
+    # Missing docstring in public method
+    D102
+    # Too many leading '#' for block comment
+    E266
```
Comment (adamgayoso) [inline .flake8:11]: @martinkim0 I added these but we should remove them and add noqa's instead of documenting everything

File: docs/conf.py [hunk 2]
```diff
@@ -125,14 +126,14 @@ def git(*args):
 try:
     git_ref = git("name-rev", "--name-only", "--no-undefined", "HEAD")
     git_ref = re.sub(r"^(remotes/[^/]+|tags)/", "", git_ref)
-except Exception:
+except Exception:  # noqa
```
Comment (adamgayoso) [inline docs/conf.py:129]: # noqa: B902

Comment (adamgayoso) [reply docs/conf.py:129]: `# noqa: B902`

**Review by martinkim0 — APPROVED** [review]

**Pull request merged**